utils/cautils/certificate_flow.go

```diff
@@ -199,15 +199,17 @@ func (f *CertificateFlow) Sign(ctx *cli.Context, token string, csr api.Certifica
 		return err
 	}
 
-	serverBlock, err := pemutil.Serialize(resp.ServerPEM.Certificate)
-	if err != nil {
-		return err
+	if resp.CertChainPEM == nil || len(resp.CertChainPEM) == 0 {
+		return errors.New("expected certificate chain in step-ca /sign response, but got <nil>")
 	}
-	caBlock, err := pemutil.Serialize(resp.CaPEM.Certificate)
-	if err != nil {
-		return err
+	var data []byte
+	for _, certPEM := range resp.CertChainPEM {
+		pemblk, err := pemutil.Serialize(certPEM)
+		if err != nil {
+			return errors.Wrap(err, "error serializing from step-ca API response")
+		}
+		data = append(data, pem.EncodeToMemory(pemblk)...)
```

Comment:
This looks ok for CertChainPEM, but if it's empty you should create the certificate using ServerPEM and CaPEM.